django/db/models/fields/related.py

```diff
@@ -1051,6 +1104,42 @@ def _check_unique(self, **kwargs):
             else []
         )
 
+    @classmethod
+    def _has_related_models_with_db_cascading(cls, model, on_delete):
+        """
+        If the ForeignKey parent has DB cascading and the Current model has non
+        db cascading return true
+        """
+        # Optimization for the case when the model does not have non-db deletion
+        if isinstance(on_delete, DatabaseOnDelete):
+            return None
+        # Fetch all the models related to the current model
+        # In other words fetch all the ForeignKey childs.
+        related_models = [
+            rel.related_model
+            for rel in model._meta.get_fields()
+            if rel.related_model
+            and not rel.auto_created
+            and hasattr(rel.related_model, "_meta")
+        ]
+        for rel_model in related_models:
+            # check through the related models
+            # if they have DB level deletion return True
+            # Our current model already has non_db cascade
+            for rel in rel_model._meta.get_fields():
+                related_on_delete = None
+                related_remote_field = rel
+                if isinstance(rel, OneToOneRel) and hasattr(rel, "on_delete"):
+                    related_on_delete = rel.on_delete
+                    related_remote_field = rel.remote_field
+                elif (
+                    isinstance(rel, ForeignKey) or isinstance(rel, OneToOneField)
+                ) and hasattr(rel.remote_field, "on_delete"):
+                    related_on_delete = rel.remote_field.on_delete
+                if isinstance(related_on_delete, DatabaseOnDelete):
+                    return related_remote_field
```

Comment:
I think this might be a bit simpler to understand:

```suggestion
                if isinstance(rel, OneToOneRel) and isinstance(rel.on_delete, DatabaseOnDelete):
                    return rel.remote_field
                elif (
                    isinstance(rel, ForeignKey) or isinstance(rel, OneToOneField)
                ) and isinstance(getattr(rel.remote_field, "on_delete", None), DatabaseOnDelete):
                    return rel
```

I was able to verify that `rel` always has an `on_delete` attribute if it is an instance of `OneToOneRel`, so we can do `rel.on_delete` here. I think we need the `getattr` for `rel.remote_field`, but it would be nice if we didn't. Maybe it would be enough to set `on_delete = None` in `Field.__init__`?

Reply:
You could also collapse `isinstance(rel, ForeignKey) or isinstance(rel, OneToOneField)` to `isinstance(rel, (ForeignKey, OneToOneField))` in the `elif` branch.

Reply:
> You could also collapse `isinstance(rel, ForeignKey) or isinstance(rel, OneToOneField)` to `isinstance(rel, (ForeignKey, OneToOneField))` in the `elif` branch.

Good point! I'll edit the suggestions.

Reply:
This seems logical. I would try put these suggestions on test soon.